### ISSUE TITLE
if track_total_hits=true then hits no data records

### ISSUE THREAD

**Issue by coding-cgs** [openobserve/openobserve#4652]
I understand that the result should be that total is the total number of statistics and hits is the list of data Instead of zo_sql_num a record
v0.12.1
<img width="1031" alt="image" src="https://github.com/user-attachments/assets/b4acfc74-383c-4036-a7b6-d2a4765d58b8">

**Comment by hengfeiyang** [comment]
Yes, we changed the behavior, if you set `track_total_hits=true` it only return total.

Earlier we support both return data and total, but in the backend actually these are two queries, one for data, one for `count(*)` and then merge the result in one API response.

Combine two queries in one API call will let the response become slow, so now, you need fire the second query  if you need the count.
